cmd/chart-repo/utils_test.go

```diff
@@ -570,3 +588,22 @@ h251U/Daz6NiQBM9AxyAw6EHm8XAZBvCuebfzyrT
 		t.Error(err)
 	}
 }
+
+var emptyRepoIndexYAMLBytes, _ = ioutil.ReadFile("testdata/empty-repo-index.yaml")
+var emptyRepoIndexYAML = string(emptyRepoIndexYAMLBytes)
+
+type emptyChartRepoHTTPClient struct{}
+
+
+func (h *emptyChartRepoHTTPClient) Do(req *http.Request) (*http.Response, error){
+	w := httptest.NewRecorder()
+	w.Write([]byte(emptyRepoIndexYAML))
+	return w.Result(), nil
+}  
+
+
+func Test_emptyChartRepo(t *testing.T) {
+	netClient = &badHTTPClient{} 
+	_, err := fetchRepoIndex(repo{URL: "https://my.examplerepo.com"})
```

Comment:
we actually need to call `syncRepo`, as this is where the error for an empty repo is found

Reply:
done